pkg/maps/ipcache/ipcache.go

```diff
@@ -7,15 +7,27 @@ import (
 	"fmt"
 	"net"
 	"net/netip"
-	"sync"
 	"unsafe"
 
 	"github.com/cilium/cilium/pkg/bpf"
 	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
 	"github.com/cilium/cilium/pkg/option"
 	"github.com/cilium/cilium/pkg/types"
 )
 
+var Cell = cell.Provide(NewMapInterface)
+
+type MapInterface interface {
+	Delete(key *Key) error
```

Comment:
Let's reproduce the original comments of the methods here. Might also be cleaner to move `Cell`, `MapInterface` and `NewMapInterface` into cell.go?